test/extended/builds/forcepull.go

```diff
@@ -38,152 +54,356 @@ while this test is running and compare results.  Restarting your docker daemon,
+				hexIDs, ierr := exutil.GetImageIDForTags(tags)
+				o.Expect(ierr).NotTo(o.HaveOccurred())
+				for i, hexID := range hexIDs {
+					g.By(fmt.Sprintf("\n FORCE PULL TEST  tag %s hex id %s ", tags[i], hexID))
+				}
+
+				// corrupt the s2i builder image
+				exutil.CorruptImage(fullName, custBldr, "s21")
+
+				// we need to plug in the dynamic values of the internal docker registry ip/port and namespace into the docker image spec
+				fpFalseS2Ipre := exutil.FixturePath("fixtures", falseS2IBC)
+				fpFalseS2Ipost := exutil.ArtifactPath(falseS2IBC)
+				err = exutil.VarSubOnFile(fpFalseS2Ipre, fpFalseS2Ipost, falseS2IBldr, fullName)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// kick off the app/lang build and verify the builder image accordingly
+				exutil.StartBuild(fpFalseS2Ipost, buildPrefix, oc)
+				exutil.WaitForBuild("FORCE PULL TEST s2i false app/lang build 1:  ", buildName, oc)
+				exutil.VerifyImagesSame(fullName, custBldr, "FORCE PULL TEST s2i false:  ")
 
 			})
 		})
 
-		g.Context("\n FORCE PULL TEST:  when s2i force pull is true and the image is bad", func() {
+		g.Context("\n FORCE PULL TEST  when s2i force pull is true and the image is bad", func() {
+			var oc = exutil.NewCLI("force-pull-s2i-true", exutil.KubeConfigPath())
+
+			g.JustBeforeEach(func() {
+				g.By("waiting for builder service account")
+				err := exutil.WaitForBuilderAccount(oc.KubeREST().ServiceAccounts(oc.Namespace()))
+				o.Expect(err).NotTo(o.HaveOccurred())
+			})
+
 			g.It("\n FORCE PULL TEST s2i true", func() {
-				fpTrueS2I := exutil.FixturePath("fixtures", "forcepull-true-s2i.json")
-				g.By(fmt.Sprintf("\n%s FORCE PULL TEST s2i true:  calling create on %s", time.Now().Format(time.RFC850), fpTrueS2I))
-				exutil.StartBuild(fpTrueS2I, buildPrefix, oc)
 
-				exutil.WaitForBuild("FORCE PULL TEST s2i true: ", buildName, oc)
+				// create the image streams
+				isTrueS2I := exutil.FixturePath("fixtures", trueS2IIS)
+				exutil.CreateResource(isTrueS2I, oc)
+
+				// kick off the build for the new builder image we can corrupt in parallel without conflicting with
+				// any other tests potentially running in parallel
+				bldrTrueS2I := exutil.FixturePath("fixtures", trueS2IBldrPrefix+".json")
+				exutil.StartBuild(bldrTrueS2I, trueS2IBldrPrefix, oc)
+				exutil.WaitForBuild("FORCE PULL TEST  s2i true bldr build:  ", trueS2IBldrPrefix+"-1", oc)
+
+				// get the auth so we can pull the build image from the internal docker registry since docker builder image will  removed it
+				// from the docker daemon cache when the docker build complete; we'll also get the internal docker registry's IP and port
+				// for pulling and tagging
+				authCfg, err := exutil.ConstructAuthCfg(oc)
+
+				// now actually pull the image back in from the openshift internal docker registry
+				fullName := authCfg.ServerAddress + "/" + oc.Namespace() + "/" + trueS2IBldr
+				err = exutil.PullImage(fullName, *authCfg)
+				o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
Yep, although I have a common oc variable for all the tests, the namespace and the contents of the .dockercfg entirely change as each `g.It({...})` is run.

There seems to be commonality between code running in a specific invocation of `g.BeforeEach` and `g.It`, but it looks like the namespace is destroyed even before the `g.AfterEach` is run:

`STEP: Destroying namespace "extended-test-forcepull-gmyey" for this suite.`

And this destroying of the namespace nukes the prior set of credentials used, *AND* any images built under that namespace, like `172.30.39.228:5000/extended-test-forcepull-gmyey/cust-forcepull-origin-custom-docker-builder`.

So, for example, the reset tags and pull images after a test that was failing in a g.AfterTest succeeded when I put that code in the g.It block.

Also, changing the image name, and replacing the namespace with something constant like "openshift" did not work from test case to test case.  So I think this confirms that any images built are nuked with the namespace, just as the credentials are totally replaced, which I confirmed but dumping the entire secret list with each BeforeEach/It invocation and seeing a completely different list each time.

So, barring additional instructions/suggestiongs, next steps:
a) I changed the code to rebuild the forcepull bldr images (exactly what we were trying to avoid) on each BeforeEach/It pairing, and thing work as expected again.
b) I could combine the 6 permutations into 1 g.It method (hence 1 namespace).  Build the bldr images once, reset the tags after each corruption is tested, etc.

It flies somewhat against the ginkgo conventions, but avoids the extra building.  I'll down that path Monday AM, unless again, someone finds fault with my investigation/results above.